server/api/redirector.go

```diff
@@ -68,16 +69,27 @@ func (h *redirector) ServeHTTP(w http.ResponseWriter, r *http.Request, next http
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
-
-	newCustomReverseProxies(urls).ServeHTTP(w, r)
+	tlsConfig, err := h.s.GetSecurityConfig().ToTLSConfig()
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	newCustomReverseProxies(urls, tlsConfig).ServeHTTP(w, r)
 }
 
 type customReverseProxies struct {
 	urls   []url.URL
 	client *http.Client
 }
 
-func newCustomReverseProxies(urls []url.URL) *customReverseProxies {
+func newCustomReverseProxies(urls []url.URL, cfg *tls.Config) *customReverseProxies {
+	dialClient = &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig:   cfg,
+			DisableKeepAlives: true,
```

Comment:
Can we enable keep alive? See https://github.com/pingcap/pd/issues/1718

Reply:
Maybe we can enable it in another PR.